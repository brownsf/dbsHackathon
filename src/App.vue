<template>
  <div id="app">
    <v-app>
      <header-comp />
      <v-content>
        <v-container>
          <router-view/>
        </v-container>
      </v-content>
    </v-app>
  </div>
</template>

<script>
import headercomp from './components/Header';

export default {
  name: 'App',
  components: {
    'header-comp': headercomp,
  },
};
</script>
