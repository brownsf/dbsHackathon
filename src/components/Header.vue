<template>
  <v-toolbar>
    <v-toolbar-title>Tech-Board</v-toolbar-title>
    <v-spacer/>
    <v-toolbar-items class="hidden-sm-and-down">
      <v-btn
        flat
        to="topics">Topics</v-btn>
      <v-btn flat>Users</v-btn>
      <v-btn
        v-if="loggedIn"
        flat
        to="login">Login</v-btn>
      <v-btn
        v-else
        flat>Logout</v-btn>
    </v-toolbar-items>
  </v-toolbar>
</template>

<script>
export default {
  computed: {
    loggedIn() {
      return this.$store.getters.isAuth;
    },
  },
};
</script>

