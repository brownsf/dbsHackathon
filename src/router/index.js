import Vue from 'vue';
import Router from 'vue-router';
import Register from '@/components/Register';
import Login from '@/components/Login';
import Topics from '@/components/Topics';
import Main from '@/components/Main';

Vue.use(Router);

export default new Router({
  routes: [
    {
      path: '/',
      name: 'Main',
      component: Main,
    },
    {
      path: '/register',
      name: 'Register',
      component: Register,
    },
    {
      path: '/login',
      name: 'Login',
      component: Login,
    },
    {
      path: '/topics',
      name: 'Topics',
      component: Topics,
    },
  ],
});
