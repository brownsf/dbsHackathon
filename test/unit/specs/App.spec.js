import { shallow } from '@vue/test-utils';
import VueRouter from 'vue-router';
import Vue from 'vue';
import Vuex from 'vuex';
import Vuetify from 'vuetify';
import App from '@/App';
import router from '@/router';
import Header from '@/components/Header';

Vue.use(Vuex);
Vue.use(Vuetify);
Vue.use(VueRouter);
describe('App', () => {
  let wrapper;
  const isAuth = jest.fn(() => false);
  const logout = jest.fn();
  beforeEach(() => {
    const state = { isAuth: false, data: {} };
    const actions = {
      register: jest.fn(),
      getAllTopics: jest.fn(),
      login: jest.fn(),
      logout,
    };
    const getters = {
      isAuth,
      getTopics: () => [{ id: 1, name: 'yep' }],
    };

    const store = new Vuex.Store({
      state,
      actions,
      getters,
    });
    wrapper = shallow(App, { store, router });
  });

  it('should render', () => {
    // expect(wrp.html()).toMatchSnapshot();

    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should have a header', () => {
    const Constructor = Vue.extend(App);
    const vm = new Constructor({ router }).$mount();
    expect(vm.$el).toMatchSnapshot();
    expect(App.components).toEqual({ 'header-comp': Header });
  });
});
