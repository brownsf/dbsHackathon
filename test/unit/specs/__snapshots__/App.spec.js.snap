// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should have a header 1`] = `
<div
  id="app"
>
  <div
    class="application theme--light"
    data-app="true"
    id="app"
  >
    <div
      class="application--wrap"
    >
      <!---->
       
      <main
        class="content"
        style="padding: 0px 0px 0px 0px;"
      >
        <div
          class="content--wrap"
        >
          <div
            class="container"
          >
            <div
              class="chart-example"
              id="chart"
            >
              <svg />
            </div>
          </div>
        </div>
      </main>
    </div>
  </div>
</div>
`;

exports[`App should render 1`] = `
<div id="app">
  <!---->
</div>
`;
